addon/index.js

```diff
@@ -25,8 +25,19 @@ window.addEventListener('storage', function ({ key, newValue }) {
   localStorageCache.set(key, jsonParseAndFreeze(newValue));
 });
 
-export default function localStorageDecorator(customLocalStorageKey) {
-  return function (target, key, descriptor) {
+export default function localStorageDecoratorFactory(...args) {
+  const isDirectDecoratorInvocation = isElementDescriptor(...args);
+  let customLocalStorageKey = null;
+
+  if (isDirectDecoratorInvocation) {
+    let [target, propertyKey, descriptor] = args;
+    return localStorageDecorator(target, propertyKey, descriptor);
+  }
+
+  customLocalStorageKey = args[0];
+  return localStorageDecorator;
```

Comment:
I would prefer a single return statement for readability.


```suggestion
  return isDirectDecoratorInvocation ? localStorageDecorator(...
args) : localStorageDecorator;
```